packages/clay-data-provider/src/ClayDataProvider.js

```diff
@@ -14,31 +13,38 @@ import templates from './ClayDataProvider.soy.js';
 class ClayDataProvider extends ClayComponent {
 	/**
 	 * Makes the request and defines initial data while it is requesting.
+	 * @param {!string} query
 	 * @param {!number} requestRetries
 	 * @protected
 	 */
-	_fetchData(requestRetries = 0) {
-		if (
-			this.initialData &&
-			!this._pollingInterval &&
-			requestRetries === 0
-		) {
+	updateData(query, requestRetries = 0) {
+		if (this._hasData(this.dataSource)) {
```

Comment:
@4lejandrito is this needed? We're already doing this in the `created` method.

Reply:
Hi @carloslancha, see https://github.com/carloslancha/clay/pull/8. With the last commit we can remove that if statement. Now we filter the items every time the query changes or every time the data changes.

Reply:
I think I accidentally pushed the commit to this PR too...